### PR TITLE
Fixes for the ImageMagick package.

### LINKED ISSUE #1694: ImageMagick fails self-tests
The ImageMagick package runs its self tests by default. These fail on Ubuntu 16.04 when using GCC 6.2, leading to a build failure.
```
PASS: Magick++/tests/tests.tap 1
PASS: Magick++/tests/tests.tap 2
PASS: Magick++/tests/tests.tap 3
PASS: Magick++/tests/tests.tap 4
PASS: Magick++/tests/tests.tap 5
PASS: Magick++/tests/tests.tap 6
PASS: Magick++/tests/tests.tap 7
PASS: Magick++/tests/tests.tap 8
PASS: Magick++/tests/tests.tap 9
FAIL: Magick++/tests/tests.tap 10
PASS: Magick++/tests/tests.tap 11
PASS: Magick++/tests/tests.tap 12
PASS: Magick++/tests/tests.tap 13
PASS: Magick++/demo/demos.tap 1
PASS: Magick++/demo/demos.tap 2
PASS: Magick++/demo/demos.tap 3
PASS: Magick++/demo/demos.tap 4
FAIL: Magick++/demo/demos.tap 5
PASS: Magick++/demo/demos.tap 6
PASS: Magick++/demo/demos.tap 7
PASS: Magick++/demo/demos.tap 8
PASS: Magick++/demo/demos.tap 9
PASS: Magick++/demo/demos.tap 10
PASS: Magick++/demo/demos.tap 11
PASS: Magick++/demo/demos.tap 12
PASS: Magick++/demo/demos.tap 13
PASS: Magick++/demo/demos.tap 14
PASS: Magick++/demo/demos.tap 15
PASS: Magick++/demo/demos.tap 16
PASS: Magick++/demo/demos.tap 17
PASS: Magick++/demo/demos.tap 18
PASS: Magick++/demo/demos.tap 19
PASS: Magick++/demo/demos.tap 20
PASS: Magick++/demo/demos.tap 21
PASS: Magick++/demo/demos.tap 22
PASS: Magick++/demo/demos.tap 23
PASS: Magick++/demo/demos.tap 24
PASS: tests/validate-formats-memory.tap 1
PASS: tests/validate-formats-disk.tap 1
PASS: tests/validate-formats-map.tap 1
============================================================================
Testsuite summary for ImageMagick 7.0.2
============================================================================
# TOTAL: 86
# PASS:  84
# SKIP:  0
# XFAIL: 0
# FAIL:  2
# XPASS: 0
# ERROR: 0
```

### PR DESCRIPTION
**Pull request by js947**
- Add pango as a dependency.
- Add (new package) ghostscript-fonts as a dependency.

Fixes #1694.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ghostscript-fonts/package.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    homepage = "http://ghostscript.com/"
+    url = "ftp://ftp.imagemagick.org/pub/ImageMagick/delegates/ghostscript-fonts-std-8.11.tar.gz"
+
+    version('8.11', '6865682b095f8c4500c54b285ff05ef6')
+
+    def install(self, spec, prefix):
+        font_dir = join_path(prefix.share, 'font')
+        mkdirp(font_dir)
+        which('install')('-t', font_dir, *os.listdir('.'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ghostscript-fonts/package.py]: Spack already has an `install()` method builtin that you can use.

**Comment by js947** [comment]
@adamjstewart As far as I could see spack's `install()` method boils down to a copy. I wanted the coreutils version which has an option `-t`.

**Comment by adamjstewart** [comment]
`install` doesn't have a `-t` argument on macOS, so that won't work. If you're trying to copy a directory, use `install_tree()`.

**Comment by js947** [comment]
#1581

**Comment by adamjstewart** [comment]
Until its documented, see current packages that use it for examples:
```console
$ grep install_tree var/spack/repos/builtin/packages/*/package.py
```

**Comment by adamjstewart** [comment]
I think all you need is something like:
```python
install_tree('fonts', font_dir)
```

**Comment by js947** [comment]
Yep! Just recompiling all the things to test it.

File: var/spack/repos/builtin/packages/ghostscript-fonts/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ghostscript-fonts/package.py]: This import isn't needed anymore.

**Comment by adamjstewart** [comment]
LGTM, thanks!

**Comment by robertdfrench** [comment]
I am setting up a new Cray XC-40 system and ImageMagick is next on my list to install. Thank you for updating this.

**Comment by js947** [comment]
Thanks for the help @adamjstewart !

**Pull request merged**